NASA ADS.js

```diff
@@ -35,131 +35,228 @@
 	***** END LICENSE BLOCK *****
 */
 
+function getSearchResults(doc) {
+	const results = doc.querySelectorAll("a[href$=abstract]");
+	const entries = {};
+	for (let el of results) {
+		const titleEl = el.querySelector(":scope h3");
+		if (!titleEl) {
+			continue;
+		}
+		const hrefParts = el.getAttribute("href").split("/");
+		if (hrefParts.length > 2) {
+			const identifier = hrefParts[hrefParts.length - 2];
+			entries[identifier] = ZU.trimInternal(titleEl.textContent);
+		}
+	}
+	return entries;
+}
 
-// attr()/text() v2
-function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null;}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null;}
-
+function extractId(url) {
+	return /\/abs\/(.*)\/abstract/.exec(url)[1];
+}
 
 function detectWeb(doc, url) {
-	if (attr(doc, 'input[name="bibcode"][type="checkbox"]', 'value')) {
+	if (url.includes("/search/")) {
 		return "multiple";
-	} else if (attr(doc, 'input[name="bibcode"][type="hidden"]', 'value')){
-		return "journalArticle";
 	}
-}
+	else if (url.includes("/abs/")) {
+		// bibcodes always start with 4 digit year, then bibstem
+		const bibstem = extractId(url).slice(4);
 
-function scrape(bibcodes, url, doc) {
-	// We can retrieve one RIS file containing the bibliographic
-	// information of multiples entries specified by their
-	// bibcode.
-	var hostname = url.replace(/\/(cgi-bin|abs)\/.*/, '');
-	var getURL = hostname + '/cgi-bin/nph-bib_query?'+bibcodes.join('&')+'&data_type=REFMAN&nocookieset=1';
-	// Some articles have a full text link to the PDF, which
-	// we can find by going through all links in <a>-tags. This
-	// works for the detail page as well as for the mulitple page.
-	var atags = doc.querySelectorAll('a');
-	Zotero.Utilities.HTTP.doGet(getURL, function(text){	
-		// load translator for RIS
-		var translator = Zotero.loadTranslator("import");
-		translator.setTranslator("32d59d2d-b65a-4da4-b0a3-bdd3cfb979e7");
-		translator.setString(text);
-		translator.setHandler("itemDone", function(obj, item) {
-			if (item.url && item.url.includes('/abs/')) {
-				let bibcode = item.url.split('/abs/')[1];
-				let pdfUrl;
-				// Go through all a tags and look for one including the
-				// corresponding bibcode and the link_type for full-text PDF.
-				for (let i=0; i<atags.length; i++) {
-					let atagUrl = atags[i].href;
-					if (atagUrl.includes('bibcode='+bibcode) && atagUrl.includes('link_type=ARTICLE')) {
-						pdfUrl = atagUrl;
-					}
-				}
-				if (pdfUrl) {
-					//Z.debug(pdfUrl);
-					item.attachments = [{
-						url: pdfUrl,
-						title: "Full Text PDF",
-						mimeType: "application/pdf"
-					}];
-				}
-			}
-			item.complete();
-		});	
-		translator.translate();
-	});
+		if (bibstem.startsWith("MsT")) {
+			return "Masters thesis";
```

Comment:
Only `thesis` is a valid Zotero item type. You could see if you can get the Master vs. Ph.D. thesis into the `type` field in Zotero

Reply:
So I have a question on this, I updated the itemType returned here to be `thesis`.

```
const bibstem = extractId(url).slice(4);
if (bibstem.startsWith("MsT") || bibstem.startsWith("PhDT")) {
  return "thesis";
}
return "journalArticle";
```

I can get the correct output: `detectWeb returned type "thesis"` when running `detectWeb` from the editor.  But when I run the `doWeb` function I always get:

```
Returned item:
{
  "itemType": "journalArticle"
  ...
}
```
Is this being set by the RIS importer? or is there something that I'm forgetting to set prior to passing that text?

Reply:
Yes, I'm pretty sure that's the RIS importer. You have 
`TY  - Thesis/Dissertation` in the RIS, which isn't valid and Zotero uses journalArticle as a generic fallback for those cases. It should instead be
`TY  - THES` It'd be best to change it in the RIS so that everyone benefits, but if that's impossible, you can overried the importers item type in the translator.

Reply:
@adam3smith So I talked with some people on the team about this, it seems like the `THES` is just an abbreviation and that the longer version is also valid (at least this is what I've heard) so I don't think it will be changing on our end.
I've updated the PR though to do it manually, hopefully I did it the right way.
Thanks!